change-address/REPWRITERSPECS/BANNO.CHANGE.ADDR.V1.POW

```diff
@@ -276,6 +306,475 @@ PRINT TITLE="BANNO.CHANGE.ADDR.V1.POW"
+       SET ZIPCODE TO UPPERCASE(NEWZIPCODE)
+      END
+
+     BNODEBUGMSG=FORMAT("CARD NAME LOC #####9 TARGETED. ",CARDLVLLOCS(TMPLOOP))+"FMERROR='"+FMERROR+"'"
+     CALL BNODEBUGMSGADD
+
+     IF FMERROR<>"" THEN
+      DO
+       RESPONSECODE=503
+       ERROR503CHR=FORMAT("Banno Chg Addr Err - Card Nm LOC #####9",CARDLVLLOCS(TMPLOOP))+" FM-"+FMERROR
+       PARSEDTEXT(1)=FORMAT("Banno Chg Addr Err - Card Nm LOC #####9",CARDLVLLOCS(TMPLOOP))
+       PARSEDTEXT(2)="FM-"+FMERROR
+       CALL CREATENOTERECORD
+      END
+    END
+  END
+
+[ External Loan level name FM]
+ IF MATCHEXTERNALNAMES=TRUE AND
+    EXTERNALNAMECOUNT>0 AND
+    RESPONSECODE=0 THEN
+  DO
+   FOR TMPLOOP=1 TO EXTERNALNAMECOUNT
+    DO
+     FMPERFORM REVISE EXTERNALLOAN LOC EXTERNALLOC(TMPLOOP) NAME LOC EXTERNALLVLLOCS(TMPLOOP)(0,0,FMERROR)
+      DO
+       SET EXTRAADDRESS TO UPPERCASE(NEWEXTRAADDR)
+       SET STREET TO UPPERCASE(NEWSTREET)
+       SET CITY TO UPPERCASE(NEWCITY)
+       SET STATE TO UPPERCASE(NEWSTATE)
+       SET ZIPCODE TO UPPERCASE(NEWZIPCODE)
+      END
+
+     BNODEBUGMSG=FORMAT("X LOAN NAME LOC #####9 TARGETED. ",EXTERNALLVLLOCS(TMPLOOP))+"FMERROR='"+FMERROR+"'"
+     CALL BNODEBUGMSGADD
+
+     IF FMERROR<>"" THEN
+      DO
+       RESPONSECODE=503
+       ERROR503CHR=FORMAT("Banno Chg Addr Err - X Ln Nm LOC #####9",EXTERNALLVLLOCS(TMPLOOP))+" FM-"+FMERROR
+       PARSEDTEXT(1)=FORMAT("Banno Chg Addr Err - X Ln Nm LOC #####9",EXTERNALLVLLOCS(TMPLOOP))
+       PARSEDTEXT(2)="FM-"+FMERROR
+       CALL CREATENOTERECORD
+      END
+    END
+  END
+ IF RESPONSECODE=0 THEN
+  DO
+   ADDRCHGSUCCESSFUL=TRUE
+   CALL POSTFMPROCESS
+  END
+END [PROCEDURE]
+
+
+
+PROCEDURE POSTFMPROCESS
+[* Clears and sets warning codes and sends email (if
+** activated) upon successful FM.
+*]
+ PARSEDTEXT(1)="Banno Successful Address Change"
+ CALL CREATENOTERECORD
+
+ IF WARNINGTOCLEAR>0 THEN
+  DO
+   FMPERFORM REVISE ACCOUNT (0,0,CLEARWARNERROR)
+    DO
+     CLEARWARNING WARNINGTOCLEAR
+    END
+   IF CLEARWARNERROR<>"" THEN
+    BNODEBUGMSG=FORMAT("WARNING #####9 UNSUCCESSFULLY CLEARED: ",WARNINGTOCLEAR)+CLEARWARNERROR
+   ELSE
+    BNODEBUGMSG=FORMAT("WARNING #####9 SUCCESSFULLY CLEARED",WARNINGTOCLEAR)
+   CALL BNODEBUGMSGADD
+  END
+ IF WARNINGTOSET>0 THEN
+  DO
+   FMPERFORM REVISE ACCOUNT (0,0,SETWARNERROR)
+    DO
+     SETWARNING WARNINGTOSET PARAMSETWARNINGEXPIRATION
+    END
+   IF SETWARNERROR<>"" THEN
+    BNODEBUGMSG=FORMAT("WARNING #####9 UNSUCCESSFULLY SET: ",WARNINGTOSET)+SETWARNERROR
+   ELSE
+    BNODEBUGMSG=FORMAT("WARNING #####9 SUCCESSFULLY SET",WARNINGTOSET)
+   CALL BNODEBUGMSGADD
+  END
+END [PROCEDURE]
+
+PROCEDURE GETNAMELOCS
+[* Collects name locators and associated identifying
+** information for allowed name types and levels. Only
+** called if old address matching is enabled.
+*]
+ IF MATCHACCTNAMES=TRUE THEN
+  DO
+   FOR EACH NAME WITH (NAME:EXPIRATIONDATE=DATENULL AND
+                       NAME:DEATHDATE=DATENULL AND
```

Comment:
I suggest all of these Each type of Name With Loops changing "(Whichever NAME:EXPIRATIONDATE=DATENULL AND" to ((Whichever NAME :EXPIRATIONDATE>SYSTEMDATE OR
                 Whichever NAME:EXPIRATIONDATE=DATENULL) AND

Reply:
You're right. Done.